src/compiler/compiler.rs

```diff
@@ -462,17 +462,26 @@ pub enum CompilerArguments<T>
     /// Commandline can be handled.
     Ok(T),
     /// Cannot cache this compilation.
-    CannotCache(&'static str),
+    CannotCache(&'static str, Option<String>),
```

Comment:
Does this buy us a whole lot? Do we wind up with anything but "failed to parse argument value" in the second field? I have a WIP patch for https://github.com/mozilla/sccache/issues/263 that saves counts of the `CannotCache` reasons in the server stats. I don't think this change will be a problem for it, but I'm not sure how I'd expose this new field.

Reply:
Ah, I think you're inverting the way you're looking at this - "argument parse" is the consistent value that always gets put in the first field (exactly because of the cannotcache reason counting), but it didn't seem ideal to just throw away all nuance of that. Specifically, take a look at `ArgParseError` - `Display` creates a description of what went wrong with what part of the argument, which isn't possible in the bounds of a static string. This nice description will get printed in the logs.